src/adapter/oso/src/dataset_resource.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::collections::HashSet;
+
+use oso::PolarClass;
+
+///////////////////////////////////////////////////////////////////////////////
+
+#[derive(PolarClass, Debug, Clone)]
+pub struct DatasetResource {
+    #[polar(attribute)]
+    pub created_by: String,
+    #[polar(attribute)]
+    pub allows_public_read: bool,
+    #[polar(attribute)]
+    pub authorized_readers: HashSet<String>,
+    #[polar(attribute)]
+    pub authorized_editors: HashSet<String>,
```

Comment:
does OSO support something like:
```rust
authorized_users: HashMap<AccountID, Role>
```
where `Role` could be `Read | Write | Admin | ...`?

I thought "Relationship Based" means an ability to annotate how use is related to the specific resource.

Reply:
`HashMap` is only supported with string keys, that's okay, accounts are strings.
It's not a trivial syntax, but I've made it working.

However, I could not make it run with the enum type, no matter what I tried.
Registering a Polar class for an enum allowed to write expressions like `Role::Read` in Polar policy, but I could not find any information how to compare the values of enum types. Naive == resulted in "unsupported operation" errors.